### PR TITLE
feat: add recommended plugins install extra

### PR DESCRIPTION
**Pull request by fubuloubu**
### What I did

Adds an install extra to install commonly used plugins to reduce installation issues

### How I did it

Add plugin packages to an install extra

### How to verify it

`pip install -U eth-ape[recommended-plugins]`

### Checklist

- [x] All changes are completed
~~- [ ] New test cases have been added~~
- [x] Documentation has been updated
- [x] Update docker container build process

### CODE REVIEW

**Review by sabotagebeats — APPROVED** [review]
tested and seems to work for me

File: setup.py [hunk 1]
```diff
@@ -46,6 +46,23 @@
         "pytest-watch",  # `ptw` test watcher/runner
         "ipdb",  # Debugger (Must use `export PYTHONBREAKPOINT=ipdb.set_trace`)
     ],
+    # NOTE: These are extras that someone can install to get up and running quickly w/ ape
+    #       They should be kept up to date with what works and what doesn't out of the box
+    #       Usage example: `pipx install eth-ape[recommended-plugins]`
+    "recommended-plugins": [
```
Comment (unparalleled-js) [inline setup.py:52]: Maybe add a part to readme about this?

File: setup.py [hunk 2]
```diff
@@ -46,6 +46,23 @@
...
+        "ape-addressbook",  # Addressbook converter
+        "ape-alchemy",  # Alchemy public network provider
+        "ape-ens",  # ENS converter
+        "ape-etherscan",  # Etherscan explorer plugin
+        "ape-foundry",  # Foundry local and fork network EVM provider
+        "ape-hardhat"  # Hardhat local and fork network EVM provider
+        "ape-infura",  # Infura public network provider
+        "ape-ledger",  # Ledger Nano S/X hardware wallet
+        "ape-notebook",  # Jupyter notebook support
+        "ape-template",  # Cookiecutter template support
```
[the remark this answers is not in the document]
Comment (unparalleled-js) [reply setup.py:62]: Oh, it does exist... Nice.

File: setup.py (hunk 1 above)
Comment (fubuloubu) [reply setup.py:52]: Yeah, I was thinking to update the install guide. Could update README too

**Comment by fubuloubu** [comment]
Built w/ Docker and seems to work!

```sh
$ docker run 2530fbf9d140 -- plugins list --all
Installed Core Plugins:
  accounts       
  test           
  geth           
  plugins        
  init           
  networks       
  run            
  console        
  ethereum       
  compile        
  pm             

Installed Plugins:
  ens            0.2.0
  hardhat        0.2.1
  alchemy        0.2.0
  infura         0.2.0
  foundry        0.2.0
  etherscan      0.2.1
  trezor         0.2.0
  template       0.2.0
  ledger         0.2.1
  addressbook    0.2.0a1
  tokens         0.2.0
  notebook       0.1.0a1

Available Plugins:
  cairo
  bsc
  ganache
  solidity
  starknet
  tenderly
  fantom
  polygon
  vyper
```

**Review by fubuloubu — COMMENTED** [review]
Note: I had removed the `setuptools` install guide, and added `pipx` install guide in this PR

**Review by sabotagebeats — APPROVED** [review]
LGTM

**Review by unparalleled-js — APPROVED** [review]
LGTM

**Pull request merged**